### ISSUE TITLE
Fix S4158 FP: Should not report on HashSet.UnionWith for readonly fields.

### ISSUE THREAD

**Issue by pavel-mikula-sonarsource** [SonarSource/sonar-dotnet#8483]
https://sonarcloud.io/project/issues?open=AYqIsHtcRlgCy-_ohqo9&id=sonarlint-visualstudio

S4158 should not report on `UnionWith`, when the symbol is a read-only field. Because changing it to non-read-only is not a desirable action.

To consider: should it be even raised on local variables? As there's no `AddRange`, and replacing the entire instance is not always an option. 

```C#
private readonly ISet<string> supportedSchemes = new HashSet<string>(StringComparer.OrdinalIgnoreCase);

public void Go(ISet<string> supportedSchemes)
{
    this.supportedSchemes.Clear();
    this.supportedSchemes.UnionWith(supportedSchemes); // FP, as there's no AddRange method in HashSet
}
```
